File: trab1/cpp/magnitude.cpp
#include <cstdlib>
#include <iostream>
#include <cmath>

using namespace std;

template <typename T>
void precisao() {
	T z, m;
	int n;

	//Algoritmo MAGNITUDE
	z = 1;// Z <- 1
	while (z > 0) {// Enquanto Z > 0 executar
		m = z;// M <- Z
		z /= 2;// Z <- Z/2
	}// Fim (Enquanto)
	n = floor(log2(abs(log(m))) + 1);// N1 <- int(log(abs(log(M)/log(2))+1)/log(2))
	cout << "Magnitude: " << m << endl;// Escrever "Magnitude:"; M
	cout << "Numero de bits do expoente: " << n << endl;// Escrever "Número de bits do expoente:"; N1
}

int main(int argc, char** argv) {
	cout << "Magnitude usando float:" << endl;
	precisao<float>();
	cout << endl;

	cout << "Magnitude usando double:" << endl;
	precisao<double>();
}

File: trab1/ruby/magnitude.rb
include Math
# Algoritmo MAGNITUDE

z = 1.0 # Z <- 1
while z > 0.0 do # Enquanto Z > 0 executar
  m = z # M <- Z
  z /= 2.0 # Z <- Z/2
end # Fim (Enquanto)
n = (log((log(m) / log(2.0)).abs + 1.0) / log(2.0)).floor # N_1 <- int(log(abs(log(M)/log(2))+1))
puts "Magnitude: #{m}" # Escrever "Magnitude:"; M
puts "Numero de bits do expoente: #{n}" # Escrever "Número de bits do expoente:"; N_1

